### PR TITLE
Add recipe for pyigrf

### PR DESCRIPTION
**Pull request by jordan-heemskerk**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml" 
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by jordan-heemskerk** [comment]
@conda-forge/help-python I think this is ready for merge

File: recipes/pyigrf/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  url: "https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz"
+  sha256: "ae9e27c6bcbb2730abd1069acd5d0ca72d8fa3d8e71ac75dc1115e52a1b196dc"
+
+build:
+  number: 0
+  script: "{{ PYTHON }} -m pip install . --no-deps --ignore-installed -vv "
+
+requirements:
+  host:
+    - numpy
```
Comment (npavlovikj) [inline recipes/pyigrf/meta.yaml]: Usually `pip` and `python` are enough for the `host` section, so please remove `numpy`.

File: recipes/pyigrf/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  imports:
+    - pyIGRF
+
+about:
+  home: "https://github.com/zzyztyy/pyIGRF"
+  license: "MIT"
+  license_family: "MIT"
+  license_file: "LICENSE"
+  summary: "IGRF-12 Model by Python"
+  doc_url: ""
```
Comment (npavlovikj) [inline recipes/pyigrf/meta.yaml]: If there is nothing to fill some fields with, you can just remove them. Please do that with `doc_url`.

File: recipes/pyigrf/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+  dev_url: ""
```
Comment (npavlovikj) [inline recipes/pyigrf/meta.yaml]: `dev_url: https://github.com/zzyztyy/pyIGRF`

File: recipes/pyigrf/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+  name: "{{ name|lower }}"
+  version: "{{ version }}"
+
+source:
+  url: "https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz"
+  sha256: "ae9e27c6bcbb2730abd1069acd5d0ca72d8fa3d8e71ac75dc1115e52a1b196dc"
+
+build:
+  number: 0
+  script: "{{ PYTHON }} -m pip install . --no-deps --ignore-installed -vv "
```
Comment (npavlovikj) [inline recipes/pyigrf/meta.yaml]: Since this it not architecture dependent package, please use `noarch: python`.

**Comment by jordan-heemskerk** [comment]
Thanks for the review @npavlovikj, I've made the requested changes

**Comment by npavlovikj** [comment]
Thank you for contributing and for the prompt fixes @jordan-heemskerk  🎉

**Pull request merged**